src/webui/api/torrentscontroller.cpp

```diff
@@ -559,12 +559,20 @@ void TorrentsController::filePrioAction()
     checkParams({"hash", "id", "priority"});
 
     const QString hash = params()["hash"];
-    int fileID = params()["id"].toInt();
-    int priority = params()["priority"].toInt();
-    BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+    const QStringList fileIDs {params()["id"].split('|')};
```

Comment:
I would change param name, but it breaks API compatibility...

Reply:
I want your thoughts: we add an optional `ids` param that takes priority over the `id` param. Or we can change the name and bump the web api version.

Reply:
We shouldn't produce incompatible changes so frequently if possible.
So maybe we should have some way to keep track of these "non-urgent" changes, and then make all the changes at once when we have the required changes?

Reply:
Maybe keep the list of postponed changes near the API version declaration, e.g. "When your changes require increasing of minor version you should also apply the following:" and then list of changes.
What do you say? @Chocobo1?

Reply:
>What do you say? @Chocobo1?

I don't think I get your idea.

However I think the commit bumping API version should also include the api code changes, so it is possible to look up by reading git log.

Reply:
>I don't think I get your idea.

@Chocobo1, take this case for example.
Meaning of param 'id' was changed. Now it provide list of file IDs. It should cause changing of param name but it breaks API compatibility. It's really bad idea to make incompatible changes too frequently.
We can proceed to use old name for some time and just mark it for future changing.
When we make changes that cannot be made in a compatible way, we can add all pending changes with them.

Reply:
>We can proceed to use old name for some time and just mark it for future changing.

I don't object the idea, however I must say from my experience, revisiting an older code after some time takes much more time for the developer and outcome is usually not good enough quality.
If @Piccirello is interested, we could try it out in this case (experiment).

Reply:
>however I must say from my experience, revisiting an older code after some time takes much more time for the developer and outcome is usually not good enough quality.

So I suggest to keep list with postponed changes. We can have some instructions, where, what and how it should be changed. Or we can even have commented changes inplace.

Well, I'm not insisting. I just thought we shouldn't break API too often.

Reply:
I would be interested, maybe maintaining a separate branch with breaking api changes that can be merged pre-minor version bump.

Reply:
>maybe maintaining a separate branch with breaking api changes that can be merged pre-minor version bump.

And have opened PR for it.
Let's try something!